pkg/operator/controller/cabundle-configmap/controller.go

```diff
@@ -0,0 +1,122 @@
+package cabundleconfigmap
+
+import (
+	"context"
+	"fmt"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// controllerName is the name to be used by the logger.
+	controllerName = "cabundle_configmap_controller"
+)
+
+var log = logf.Logger.WithName(controllerName)
+
+// Config holds all the things necessary for the controller to run.
+type Config struct {
+	ServiceCAConfigMapName types.NamespacedName
+	AdminCAConfigMapName   types.NamespacedName
+	IngressCAConfigMapName types.NamespacedName
```

Comment:
Since none of these resources are consumed by the controller, but makes it available for others, thought to keep it configurable so that it could be updated as required by the caller.